Python/optimizer.c

```diff
@@ -540,30 +548,22 @@ translate_bytecode_to_trace(
                 goto done;
             }
         }
+        assert(opcode != ENTER_EXECUTOR && opcode != EXTENDED_ARG);
 
-        if (opcode == ENTER_EXECUTOR) {
-            _PyExecutorObject *executor =
-                (_PyExecutorObject *)code->co_executors->executors[oparg&255];
-            opcode = executor->vm_data.opcode;
-            DPRINTF(2, "  * ENTER_EXECUTOR -> %s\n",  _PyOpcode_OpName[opcode]);
-            oparg = (oparg & 0xffffff00) | executor->vm_data.oparg;
-        }
-
+        /* Special case the first instruction,
+         * so that we can guarantee forward progress */
         if (progress_needed) {
             progress_needed = false;
-            /* Special case the first instruction, so that we can guarantee
-            * forward progress */
-            if (opcode == JUMP_BACKWARD) {
-                instr += 2 - (int32_t)oparg;
+            if (opcode == JUMP_BACKWARD || opcode == JUMP_BACKWARD_NO_INTERRUPT) {
+                instr += 1 + _PyOpcode_Caches[opcode] - (int32_t)oparg;
+                initial_instr = instr;
```

Comment:
Interesting. This will affect some of the debug output, notably the "Created a trace for ..." message below at line 676 will mention a different byte offset than the "Optimizing ..." above at line 405. I honestly like what you set it to here better, and debug level 4 is excessively verbose, so I don't care, but some bit of me will miss the correspondence. :-)